include/aws/crt/Optional.h

```diff
@@ -196,7 +197,7 @@ namespace Aws
             }
 
           private:
-            typename std::aligned_storage<sizeof(T)>::type m_storage;
+            alignas(T) std::array<char, sizeof(T)> m_storage;
```

Comment:
Seems weird to use std::array, when we could use a simple literal array, as suggested in [P1413R3](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2021/p1413r3.pdf#page=5):

```suggestion
            alignas(T) char m_storage[sizeof(T)];
```